test/Benchmarks/src/Vector/Map_Error.enso

```diff
@@ -0,0 +1,59 @@
+from Standard.Base import all
+
+import Standard.Base.Errors.Illegal_Argument.Illegal_Argument
+from Standard.Base.Data.Array_Proxy import Array_Proxy
+
+from Standard.Test import Bench
+
+import project.Vector.Utils
+
+polyglot java import java.util.ArrayList
+
+options = Bench.options . set_warmup (Bench.phase_conf 1 3) . set_measure (Bench.phase_conf 1 3)
+
+## Repeats 0..9
+create_vec = Vector.new 100000 (i-> i%10)
+
+id i = i
+
+id_one_error i = if i != 5 then i else
+    Error.throw (Illegal_Argument.Error "error")
+
+always_throw i =
+    _ = [i]
+    Error.throw (Illegal_Argument.Error "error")
+
+throw_ten_percent i = if i > 0 then i else
+    Error.throw (Illegal_Argument.Error "error")
+
+collect_benches = Bench.build builder->
+    unaltered_data = create_vec
+
+    type_group vector_type alter =
+        data = alter unaltered_data
```

Comment:
This vector is constructed eagerly.
I don't know how long that takes - I assume allocating 100k elements is relatively fast, but also a non-zero cost.

This will be run every time our benchmarks structure is initialized, even if the particular benchmark is not scheduled to be run. While not a big issue, I think we should make sure such costly initialization is done lazily to avoid slowing down benchmark structure preparation.

You can see the pattern in benchmarks like `Table/Add_Row_Number.enso`, `Table/Aggregate.enso`, `Text/Compare.enso` and most of our benchmarks actually - we have a `type Data` that contains lazy `~fields` for the costly-to-construct data. This ensures that they are only computed when the benchmarks actually run (they then are computed on the first warmup run that uses them).

Reply:
Done.